packages/expo-dev-menu/android/build.gradle

```diff
@@ -1,10 +1,124 @@
+import groovy.json.JsonSlurper
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: 'maven'
 
 group = 'host.exp.exponent'
 version = '0.8.1'
 
+// reanimated v2
+def reactNativeFilePath = ["node", "--print", "require.resolve('react-native/package.json')"].execute().text.trim()
+def inputFile = new File(reactNativeFilePath)
+def json = new JsonSlurper().parseText(inputFile.text)
+def reactNativeVersion = json.version as String
+def (major, minor, patch) = reactNativeVersion.tokenize('.')
+
+def engine = "jsc"
+
+// If the user has the reanimated 2.2.1 or above installed, we don't have to swap JNI implementation.
+def shouldSwapJNI() {
+  def reanimatedPackagePath = ["node", "--print", "require.resolve('react-native-reanimated/package.json')"].execute().text.trim()
+  if (!reanimatedPackagePath) {
+    return true
+  }
+
+  def inputFile = new File(reanimatedPackagePath)
+  def json = new JsonSlurper().parseText(inputFile.text)
+  def reanimatedVersion = json.version as String
+  def (int major, int minor, int path) = reanimatedVersion.tokenize('.').collect { it.toInteger() }
+
+  if (major < 2 || (major == 2 && minor <= 2 && path == 0)) {
```

Comment:
```suggestion
  if (major < 2 || (major == 2 && minor <= 2 && patch == 0)) {
```

Reply:
also i think this will be a false miss for versions 2.1.x where x != 0 (if any such version exists)